src/arraymath.jl

```diff
@@ -36,6 +36,7 @@ Base.promote_op{Op,T<:Number,A<:StaticArray}(op::Op, ::Type{T}, ::Type{A}) = sim
         $(Expr(:call, SA, Expr(:tuple, v...)))
     end
 end
+Base.zero{SA <: StaticArray}(a::Union{SA,Type{SA}}) = zeros(a)
```

Comment:
I've been adding `@inline` explicitly everywhere since I'm not completely sure the heuristic inlining will work or not after several layers of inlining.